numpy/lib/tests/test_function_base.py

```diff
@@ -902,9 +902,9 @@ def test_specific_axes(self):
         # test maximal number of varargs
         assert_raises(TypeError, gradient, x, 1, 2, axis=1)
 
-        assert_raises(ValueError, gradient, x, axis=3)
-        assert_raises(ValueError, gradient, x, axis=-3)
-        assert_raises(TypeError, gradient, x, axis=[1,])
+        assert_raises(np.AxisError, gradient, x, axis=3)
```

Comment:
These exceptions were found by temporarily removing the base classes from `AxisError`, so that tests for `ValueError` and `IndexError` would not catch them.